Guide/DynamicProgramming/README.md

```diff
@@ -0,0 +1,12 @@
+# Dynamic Programming Guide
+
+## Overview
+<p> Dynamic Programming - An algorithm that solves smaller problems that can
```

Comment:
It does sound better. The explanation is more in-depth. My main nitpick is that you used the words "solve" and "problem" a lot in the first sentence.